packages/web-client/src/helpers/space/functions.ts

```diff
@@ -169,20 +169,20 @@ export function buildSpace(data): SpaceResource {
     canDownload: function () {
       return true
     },
-    canBeDeleted: function ({ user }: { user?: User } = {}) {
-      return this.disabled && (user?.role?.name === 'spaceadmin' || this.isManager(user))
+    canBeDeleted: function ({ user, ability }: { user?: User; ability?: any } = {}) {
+      return this.disabled && (ability?.can('delete', 'Space') || this.isManager(user))
```

Comment:
I don't really like that we mix the user- and the resource-permission-layers here. IMO this place should be resource-related only.

But I don't know an elegant way to separate this without refactoring more code.

Reply:
If we don't use the permissions here I only see the option to stick to the space role in the resource (i.e. space admin can't do anything if he/she is only viewer in the space) and then let the space admin use the `admin-settings` app for administrative tasks. But then again, I don't know if it would feel inconsistent to be able to do stuff in the `admin-settings` app but not in the files app.